modules/terraform/plan_struct.go

```diff
@@ -0,0 +1,108 @@
+package terraform
+
+import (
+	"encoding/json"
+
+	"github.com/gruntwork-io/terratest/modules/testing"
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// PlanStruct is a Go Struct representation of the plan object returned from Terraform. Unlike the raw plan representation
+// returned by terraform-json, this struct provides a map that maps the resource addresses to the changes and planned
+// values to make it easier to navigate the raw plan struct.
+type PlanStruct struct {
+	RawPlan tfjson.Plan
+
+	ResourcePlannedValuesMap map[string]*tfjson.StateResource
+	ResourceChangesMap       map[string]*tfjson.ResourceChange
```

Comment:
NIT: consider adding comments to what each of these fields represents / contains.

Reply:
Added comments in 77d1f0b8